### ISSUE TITLE
getMutableAIState(key [keyof AI Context]) update() and done() didn't change object reference

### ISSUE THREAD

**Issue by btxakita** [vercel/ai#2214]
### Description

`getMutableAIState` function provided a way to only mutate a specific `key` from the AIState, which is very convenient. Say we have a "messages" in the AIState object. You could use `getMutableAIState<AI>("messages").`

But when you call update() / done(). Things went out of control:

- In done() function, it use `jsondiffpatch` to find the differences between `originalState` and `currentState` from the `store` object, while when we update the object's specific key, these two states are **always the same** object. So the function just cannot find the delta.

Which then leads to other problems on client side context updates etc...

### Code example

`packages/core/rsc/ai-state.tsx`
```
// store object
return asyncAIStateStorage.run(
    {
      currentState: `state`,
      originalState: `state`,
      sealed: false,
      options,
    },
    fn,
  );`

// done() function 
const delta = jsondiffpatch.diff(`store.originalState`, `store.currentState`);
store.mutationDeltaResolve!(delta);
```
### Additional context

Also, there're many type errors regarding this scenario.

**Comment by jeremyphilemon** [comment]
Closed by #2821

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The issue reports a concrete failure in existing keyed AI state mutation behavior: updates are not detected or propagated because state references remain identical.

### Evidence
- Calling `getMutableAIState<AI>(key).update()` or `.done()` leaves `originalState` and `currentState` referencing the same object.
- Because `jsondiffpatch.diff` receives identical state objects, it cannot detect a mutation delta, causing client-side context updates to fail.
- A maintainer comment states the issue was closed by PR #2821.
